test/lint/lint-circular-dependencies.sh

```diff
@@ -11,6 +11,7 @@ export LC_ALL=C
 EXPECTED_CIRCULAR_DEPENDENCIES=(
     "chainparamsbase -> util/system -> chainparamsbase"
     "index/txindex -> validation -> index/txindex"
+    "index/blockfilterindex -> validation -> index/blockfilterindex"
```

Comment:
bf744288bd58718844841c088c498cd57a8f87d8

An alternative is to add a global `g_delay_pruning` in validation.h which gets updated every time an index `m_synced` is updated.

Reply:
Or move the "load block" function into a separate "storage" module.

Reply:
The global `g_delay_pruning` smells fragile. Better not do that.

I like the "storage" module approach which though would be something for another PR.